WordPress/openverse-frontend
# Add VFilterButton

From Zack:

Building on Olga's work and with confirmation from @panchovm I was able to simplify the state management here considerable and I also believe all the varied states are correct. There's like 5-6 permutations _per_ viewport and getting them right was intense!

The biggest way this was simplified, was previously it was attempting to turn the filter button into the close button for the mobile menu. 

![CleanShot 2021-12-15 at 17 41 56@2x](https://user-images.githubusercontent.com/6351754/146276191-02c0b987-0af5-4b47-bba6-a93c053464ec.png)

This seems natural given the positioning, but ultimately really added an extreme level of complexity to this component. Instead, I see this Openverse logo and close button in the above screenshot as _contents of the mobile menu_, which appear  _overlapping_ the mobile header. This allows us to keep the close button inside the mobile menu, and keep this component in scope. 

Other changes here that _are_ outside the scope, that I apologize for:

- Header padding optimized
- Logo size fixed to match mockups
- Header switched to sticky instead of fixed
- Hide filter button on the homepage

A note: test coverage only tests for label and icon usage, _not_ the button variation and appearance. I think it's sufficient coverage; you may disagree!

> Original message:
> ## Fixes
> 
> 
> Fixes #476 by @obulat
> 
> ## Description
> 
> 
> This PR adds the `VFilterButton` component in `VHeader`.  This is a button that opens and closes the `FiltersSidebar`, and also shows the number of applied filters on mobile screens.
> 
> ## Testing Instructions
> 
> Run `npm run dev`, and open http://localhost:8443/search?q=cat. You should see a Header with only the `VFilterButton` component. You can open and close the `FiltersSidebar` clicking on it, and if you narrow your screen, it also shows you the number of filters applied.
> 
> ## Outstanding Todos:
> - [x] Improve the styles (sizes, paddings, check for active/hover states).
> - [x] Improve the docstrings.
> - [ ] Improve the Storybook story.
> - [x] Add tests.
> 
> ## Checklist
> 
> - [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
> - [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
> - [x] My commit messages follow [best practices][best_practices].
> - [x] My code follows the established code style of the repository.
> - [x] I added or updated tests for the changes I made (if applicable).
> - [ ] I added or updated documentation (if applicable).
> - [x] I tried running the project locally and verified that there are no visible errors.
> 

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
